deps/rabbit/src/rabbit_channel_tracking.erl

```diff
@@ -178,109 +166,52 @@ shutdown_tracked_items(TrackedItems, _Args) ->
 list() ->
     lists:foldl(
       fun (Node, Acc) ->
-              Tab = tracked_channel_table_name_for(Node),
-              try
-                  Acc ++
-                  mnesia:dirty_match_object(Tab, #tracked_channel{_ = '_'})
-              catch
-                  exit:{aborted, {no_exists, [Tab, _]}} ->
-                      %% The table might not exist yet (or is already gone)
-                      %% between the time rabbit_nodes:all_running() runs and
-                      %% returns a specific node, and
-                      %% mnesia:dirty_match_object() is called for that node's
-                      %% table.
-                      Acc
-              end
+              list_on_node(Node) ++ Acc
       end, [], rabbit_nodes:all_running()).
 
 -spec list_of_user(rabbit_types:username()) -> [rabbit_types:tracked_channel()].
 
 list_of_user(Username) ->
     rabbit_tracking:match_tracked_items(
-        fun tracked_channel_table_name_for/1,
-        #tracked_channel{username = Username, _ = '_'}).
+      ?TRACKED_CHANNEL_TABLE,
+      #tracked_channel{username = Username, _ = '_'}).
 
 -spec list_on_node(node()) -> [rabbit_types:tracked_channel()].
 
+list_on_node(Node) when Node == node() ->
+    ets:tab2list(?TRACKED_CHANNEL_TABLE);
 list_on_node(Node) ->
-    try mnesia:dirty_match_object(
-          tracked_channel_table_name_for(Node),
-          #tracked_channel{_ = '_'})
-    catch exit:{aborted, {no_exists, _}} -> []
+    case rabbit_misc:rpc_call(Node, ets, tab2list, [?TRACKED_CHANNEL_TABLE]) of
```

Comment:
I wonder if it would make future changes easier if the rpc call would call the API of rabbit_channel_tracking this was not exposing the implementation detail of ets? This way in a mixed-version cluster each node can have slight differences how they implement listing/counting. A pattern similar to `rabbit_channel:list/list_local` https://github.com/rabbitmq/rabbitmq-server/blob/master/deps/rabbit/src/rabbit_channel.erl#L373-L380
(just as a dummy example in that case renaming the ets table would not require upgrade code or feature flag).

This suggestion is valid for all rpcs made in conn/channel tracking but is beyond the scope of the performance optimisation though.

Reply:
I agree